update_kb_redis.py

```diff
@@ -19,18 +19,49 @@
 error_log = []
 
 # Regexes
-enum_re = re.compile(r'enum[^{]*[^}]*')
+enum_re = re.compile(r'enum[^{]*{[^}]')
 keymap_re = re.compile(r'constuint[0-9]*_t[PROGMEM]*keymaps[^;]*')
 layers_re = re.compile(r'\[[^\]]*]=[0-9A-Z_]*\([^[]*\)')
 layout_macro_re = re.compile(r']=(LAYOUT[0-9a-z_]*)\(')
 keymap_macro_re = re.compile(r']=(KEYMAP[0-9a-z_]*)\(')
 
 # Processors
 ARM_PROCESSORS = 'cortex-m0', 'cortex-m0plus', 'cortex-m3', 'cortex-m4', 'STM32F042', 'STM32F072', 'STM32F303'
```

Comment:
why does AVR get a None but ARM does not?

Reply:
Because AVR is the default, so we match against it when we don't know.